### ISSUE TITLE
exception thrown in constraint

### ISSUE THREAD

**Issue by joeel1** [C2FO/nools#187]
just noticed, error happend during constraint's runtime evalution will crash nools, and no being handle by error callback, supplied through 'session.match'.

use case:
i)   normal constraint - name: Fact name.valid_param == "some value";
     -> matched normally
ii)  constraint with undefined/null parameters- name: Fact name.<undefined parameter> == "some value";
     -> matched normally
iii) constraint with undefined property of undefined object: Fact name[undefined parameter][undefined child] == "some value";
    -> nools crash....

any idea?.

**Comment by DevSide** [comment]
Yes, you can't use two levels of undefined keys. You can define new facts to represent intermediate data.

**Comment by joeel1** [comment]
we are interested on taking over nools, in case no one else interested on maintaining the project.
how shall we proceed?

**Comment by doug-martin** [comment]
Hi @joeel1 I have moved the nools codebase to a new org `noolsjs`. I am actively working on an [update to nools](https://github.com/doug-martin/nools/tree/es6) with the first order of business being updating the code base to es6 and to support the latest versions of node. I plan on publishing the new release in a couple of weeks once I have all the documentation updated. 

Once I finish the first update I will begin working on getting the outstanding PRs merged then Ill start working on new features. If you are interested in contributing please feel free to issue PRs against [this branch](https://github.com/doug-martin/nools/tree/es6).
